be/src/exec/pipeline/olap_chunk_source.cpp

```diff
@@ -294,15 +294,15 @@ StatusOr<vectorized::ChunkPtr> OlapChunkSource::get_next_chunk_from_buffer() {
     return chunk;
 }
 
-Status OlapChunkSource::buffer_next_batch_chunks_blocking(size_t batch_size, bool& can_finish) {
+Status OlapChunkSource::buffer_next_batch_chunks_blocking(size_t chunk_size, bool& can_finish) {
     if (!_status.ok()) {
         return _status;
     }
     using namespace vectorized;
 
-    for (size_t i = 0; i < batch_size && !can_finish; ++i) {
+    for (size_t i = 0; i < chunk_size && !can_finish; ++i) {
         ChunkUniquePtr chunk(
-                ChunkHelper::new_chunk_pooled(_prj_iter->encoded_schema(), config::vector_chunk_size, true));
+                ChunkHelper::new_chunk_pooled(_prj_iter->encoded_schema(), _runtime_state->chunk_size(), true));
```

Comment:
What is the difference between these two chunk_size ?  _runtime_state->chunk_size() vs chunk_size

Reply:
num_chunks ?

Reply:
_runtime_state->chunk_size() is depend on execute engine, pipeline is 16384, non-pipeline is 4096, this value is used to reserve column's size and used to reclaim column as an upper limit size.